### PR TITLE
Update to 2022.12.12 and automated update workflow

### PR DESCRIPTION
**Pull request by fflorent**
- Update to latest version
- Add github workflow in order to create PR on new release (run every ~15 days, because there are lots of releases)

## PR Status

- [x] Code finished and ready to be reviewed/tested
- [x] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by fflorent** [comment]
!testme

**Comment by fflorent** [comment]
NB: in order to test manually the `updater.sh` script:
1. Edit the version in `manifest.json`
2. Reset the src files to the version in master
3. Run this command: `GITHUB_ENV=/tmp/minio.env GITHUB_REPOSITORY=Yunohost-Apps/minio_ynh bash -x ./.github/workflows/updater.sh`

**Comment by fflorent** [comment]
!testme

**Comment by fflorent** [comment]
!testme

File: .github/workflows/updater.sh [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+# Any action on the app's source code can be done.
+# The GitHub Action workflow takes care of committing all changes after this script ends.
+
+#=================================================
+# GENERIC FINALIZATION
+#=================================================
+
+# Replace new version in manifest
+echo "$(jq -s --indent 4 ".[] | .version = \"$version~ynh1\"" manifest.json)" > manifest.json
```
Comment (tituspijean) [inline .github/workflows/updater.sh:131]:
```suggestion
version=$(sed -E "s/RELEASE\.([0-9-]+)T.+/\1/g" <<< $version | sed -E "s/-/./g")
echo "$(jq -s --indent 4 ".[] | .version = \"$version~ynh1\"" manifest.json)" > manifest.json
```

Cleans up the version for the manifest.

File: .github/workflows/updater.sh [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  "amd64")
+    asset_url="https://dl.min.io/server/minio/release/linux-amd64/archive/minio.$version"
+    src_filename=minio
+    ;;
+  "arm64")
+    asset_url="https://dl.min.io/server/minio/release/linux-arm64/archive/minio.$version"
+    src_filename=minio
+    ;;
+  "mc_amd64")
+    asset_url="https://dl.min.io/client/mc/release/linux-arm64/archive/mc.$mc_version"
```
Comment (tituspijean) [inline .github/workflows/updater.sh]:
```suggestion
    asset_url="https://dl.min.io/client/mc/release/linux-amd64/archive/mc.$mc_version"
```

Typo

File: conf/mc_amd64.src [hunk 3]
```diff
@@ -1,5 +1,5 @@
-SOURCE_URL=https://dl.min.io/client/mc/release/linux-amd64/archive/mc.RELEASE.2022-11-07T23-47-39Z
-SOURCE_SUM=85ad97a2ceea6b54fc856293bffe964f56c362b747b5b9d07cccfaae5c746ffa
+SOURCE_URL=https://dl.min.io/client/mc/release/linux-arm64/archive/mc.RELEASE.2022-12-13T00-23-28Z
+SOURCE_SUM=d2bc38c5173900b88eb61f4adabdf13b72aabd25d920369e5d1b48ccd6fcf6aa
```
Comment (tituspijean) [inline conf/mc_amd64.src]:
```suggestion
SOURCE_URL=https://dl.min.io/client/mc/release/linux-amd64/archive/mc.RELEASE.2022-12-13T00-23-28Z
SOURCE_SUM=47069f02014eb666680720cc7982d0fcbe96b4878abfd0139b0f0cb39b529874
```

Fixes consequence of typo in updater script

File: manifest.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
         "en": "High Performance, Kubernetes Native Object Storage",
         "fr": "Serveur de stockage d'objets hautes performances"
     },
-    "version": "2022.11.11~ynh1",
+    "version": "RELEASE.2022-12-12T19-27-27Z~ynh1",
```
Comment (tituspijean) [inline manifest.json]:
```suggestion
    "version": "2022.12.12~ynh1",
```

Fix version format

**Comment by tituspijean** [comment]
!testme

File: .github/workflows/updater.sh (hunk 2 above)
Comment (fflorent) [reply .github/workflows/updater.sh]: @tituspijean Thank you very much for this fix!

**Comment by fflorent** [comment]
Sooo, may this be merged? :)

**Comment by Limezy** [comment]
Wow, I'm so thankful for all the work done !
I was very busy these days (5th child is coming 👍) so didn't really have any time.
Let me try if that's working and merge

**Comment by Limezy** [comment]
The update is working well, but MinIO by itself isn't... 😢 
This is because of a recent change on MinIO side. I have to dig into the details to understand how to change the package accordingly

**Comment by fflorent** [comment]
Yeah, the upgrade does fix the issue #33, but a fresh installation at least should work.

There are some migration procedure, but I don't have much time to work on it at the moment.

**Comment by Limezy** [comment]
https://min.io/docs/minio/linux/operations/install-deploy-manage/migrate-fs-gateway.html
Here is a procedure. Question is how hard to run it within the package upgrade script...

**Comment by Limezy** [comment]
Migration is working on my test server.
I'm creating another PR including commits done here.
Will close that one once #35 merged

**Pull request merged**